File: sortVowelsInAString.py
class Solution:
    def sortVowels(self, s: str) -> str:
        vowels = set('aeiouAEIOU')
        s_list = list(s)

        # Extract and sort the vowels
        vowel_chars = sorted([c for c in s if c in vowels])
        
        # Replace vowels in-place in order
        vowel_index = 0
        for i in range(len(s_list)):
            if s_list[i] in vowels:
                s_list[i] = vowel_chars[vowel_index]
                vowel_index += 1

        return ''.join(s_list)

# usage/checking output
sol = Solution()
print(sol.sortVowels("lEetcOde"))  # Output: "lEOtcede"
print(sol.sortVowels("lYmpH"))     # Output: "lYmpH"

File: sortVowelsInAString.go
package main

import (
	"fmt"
	"sort"
	"strings"
)

func sortVowels(s string) string {
	vowelsSet := "aeiouAEIOU"
	isVowel := func(c byte) bool {
		return strings.ContainsRune(vowelsSet, rune(c))
	}

	// Extract vowels
	var vowels []byte
	for i := 0; i < len(s); i++ {
		if isVowel(s[i]) {
			vowels = append(vowels, s[i])
		}
	}

	// Sort vowels by ASCII
	sort.Slice(vowels, func(i, j int) bool {
		return vowels[i] < vowels[j]
	})

	// Build result
	result := []byte(s)
	vowelIdx := 0
	for i := 0; i < len(result); i++ {
		if isVowel(result[i]) {
			result[i] = vowels[vowelIdx]
			vowelIdx++
		}
	}

	return string(result)
}

/*
func main() {
	fmt.Println(sortVowels("lEetcOde")) // Output: "lEOtcede"
	fmt.Println(sortVowels("lYmpH"))    // Output: "lYmpH"
}
*/
